File: src/test/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/importering/autoimport/ImporteringServiceTest.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.importering.autoimport;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.Arrays;
import java.util.List;

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.importering.SykefraværsstatistikkImporteringService;
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domene.ÅrstallOgKvartal;
import org.junit.jupiter.api.Test;
import org.springframework.core.env.Environment;

public class ImporteringServiceTest {

  Environment environment;
  SykefraværsstatistikkImporteringService importeringService =
      new SykefraværsstatistikkImporteringService(null, null, null, true, null);

  @Test
  public void
      kanImportStartes__returnerer_TRUE_dersom_alle_årstall_og_kvartal_er_like_OG_sykefraværsstatistikk_ligger_ett_kvartal_bak_Dvh() {
    assertFalse(
        importeringService.kanImportStartes(
            List.of(new ÅrstallOgKvartal(2019, 3)), List.of(new ÅrstallOgKvartal(2020, 1))));

    assertFalse(
        importeringService.kanImportStartes(
            Arrays.asList(new ÅrstallOgKvartal(2019, 4), new ÅrstallOgKvartal(2019, 3)),
            Arrays.asList(new ÅrstallOgKvartal(2020, 1), new ÅrstallOgKvartal(2020, 1))));

    assertTrue(
        importeringService.kanImportStartes(
            Arrays.asList(new ÅrstallOgKvartal(2019, 4), new ÅrstallOgKvartal(2019, 4)),
            Arrays.asList(new ÅrstallOgKvartal(2020, 1), new ÅrstallOgKvartal(2020, 1))));
  }
}

File: src/test/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/importering/autoimport/ImporteringServiceTest.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.importering.autoimport

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller.ÅrstallOgKvartal
import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.importering.SykefraværsstatistikkImporteringService
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock

class ImporteringServiceTest {
    var importeringService = SykefraværsstatistikkImporteringService(mock(), mock(), mock(),  mock())

    @Test
    fun kanImportStartes__returnerer_TRUE_dersom_alle_årstall_og_kvartal_er_like_OG_sykefraværsstatistikk_ligger_ett_kvartal_bak_Dvh() {
        Assertions.assertFalse(
            importeringService.kanImportStartes(
                listOf(ÅrstallOgKvartal(2019, 3)), listOf(ÅrstallOgKvartal(2020, 1))
            )
        )
        Assertions.assertFalse(
            importeringService.kanImportStartes(
                listOf(ÅrstallOgKvartal(2019, 4), ÅrstallOgKvartal(2019, 3)),
                listOf(ÅrstallOgKvartal(2020, 1), ÅrstallOgKvartal(2020, 1))
            )
        )
        Assertions.assertTrue(
            importeringService.kanImportStartes(
                listOf(ÅrstallOgKvartal(2019, 4), ÅrstallOgKvartal(2019, 4)),
                listOf(ÅrstallOgKvartal(2020, 1), ÅrstallOgKvartal(2020, 1))
            )
        )
    }
}
